src/main.rs

```diff
@@ -63,19 +62,44 @@ impl<T: Handler> Haematite<T> {
 }
 
 fn main() {
+    let mut args = std::env::args();
+    args.next(); // skip the executable name
+
+    let arg = args.next();
+
+    if arg == None {
+        eprintln!("syntax: haematite <config>");
+        std::process::exit(1);
+    }
+
+    let config = match config::Config::load_from_file(arg.unwrap()) {
+        Ok(it) => it,
+        Err(err) => {
+            eprintln!("failed to read config file: {}", err);
+            std::process::exit(1);
+        }
+    };
+
     let mut haematite = Haematite::new(
         Server {
-            sid: "111".to_string(),
-            name: String::from("haematite.vpn.lolnerd.net"),
-            description: String::from("haematite psuedoserver"),
+            sid: config.sid,
+            name: config.server_name,
+            description: config.server_description,
             ..Server::default()
         },
         TS6Handler::new(),
     );
 
-    let socket = TcpStream::connect("husky.vpn.lolnerd.net:6667").expect("failed to connect");
+    let socket = TcpStream::connect(format!(
+        "{}:{}",
+        config.uplink_remote_address, config.uplink_remote_port
+    ))
```

Comment:
You actually don't need this `format`. `TcpStream::connect((config.uplink_remote_address, config.uplink_remote_port))` will also do. I know, it's not exactly obvious from the documentation.